presto-main/src/main/java/com/facebook/presto/sql/analyzer/MaterializedViewQueryOptimizer.java

```diff
@@ -585,17 +596,136 @@ protected Node visitExpression(Expression node, Void context)
             return super.visitExpression(removeExpressionPrefix(node, removablePrefix), context);
         }
 
+        class NonAssociativeFunctionHandler
+        {
+            // TODO: Move COUNT to this handler
+            // TODO: Add many more functions
+
+            private final Map<QualifiedName, Function<Identifier, Boolean>> validateFunctions = ImmutableMap.of(
```

Comment:
Create an interface with 2 functions:

`boolean validate(Identifier)`

`Expression rewrite(Identifier)`


so the map can be `Map<QualifiedName, interface>`